src/Multipart/UploadState.php

```diff
@@ -13,6 +13,18 @@ class UploadState
     const INITIATED = 1;
     const COMPLETED = 2;
 
+    protected $progressBar = [
```

Comment:
$progressBar and $progressThresholds should be combined into one array, with the keys being the thresholds and the values being what should be printed out.  One way of doing this would be array_combine.  Example:
[
0 => "Transfer initiated...\n|                    | 0.0%\n",
125 => "... 12.5%..."
//etc...
]